### PR TITLE
[tflite] Revert #2135 and add ssat tests for multi-tensor output tflite model

### PR DESCRIPTION
**Pull request by anyj0527**
This reverts PR #2135 

The location of input/output buffers are not static when the tflite model has multi-tensor output. Thus #2135 causes regressions described in #2145 (it does not happen when the model has a one output tensor)

And, add a fail case and a simple run case for multi-tensor output tflite model.

Resolves: #2145 

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

**Comment by anyj0527** [comment]
@myungjoo My appologies for this late response.

Q1: Well, while ssat test runs, the added positive testcase is failed if the commit is not reverted. But build is successful. Is there a way to stop a build process if there are certain failed ssat tests?

Q2: No I can't. All the tflite models in the directory has only one output tensor and can not catch the bug.

**Comment by anyj0527** [comment]
@myungjoo Oh about Q1,
`gbs build --define "unit_test 1"` failed. If it is not reverted.

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc [hunk 1]
```diff
@@ -183,24 +181,36 @@ TFLiteInterpreter::invoke (const GstTensorMemory * input,
...
-    output_tensor_ptr[i]->data.raw = (char *) output[i].data;
+    tensor_ptr->data.raw = (char *) input[i].data;
+    tensors_idx.push_back (tensor_idx);
   }
 
   if (!size_mismatch) {
 #ifdef ENABLE_TFLITE_NNAPI_DELEGATE
     if (use_nnapi)
-      status = nnfw_delegate->Invoke (interpreter.get ());
+      status = nnfw_delegate->Invoke (interpreter.get());
```
Comment (helloahn) [inline ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc]: this style change is not the recommended way.

Comment (anyj0527) [reply ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc]: Thank you for pointing out. It's fixed.

**Comment by wooksong** [comment]
@myungjoo @anyj0527 
FYI, 
## Before applying this PR
#### Unit Test Cases
```bash
./tests/ssat $
...omission...
==================================================
[PASSED] nnstreamer_converter (28 passed among 28 cases)
[PASSED] nnstreamer_filter_caffe2 (2 passed among 2 cases)
[PASSED] nnstreamer_mux (84 passed among 84 cases)
[PASSED] nnstreamer_filter_tensorflow_lite (21 passed among 21 cases)
[PASSED] transform_stand (3 passed among 3 cases)
[PASSED] nnstreamer_decoder_image_segment (4 passed among 4 cases)
[PASSED] nnstreamer_split (21 passed among 21 cases)
[PASSED] nnstreamer_decoder_boundingbox (6 passed among 6 cases)
[PASSED] nnstreamer_filter_python (6 passed among 6 cases)
[PASSED] nnstreamer_repo_rnn (2 passed among 2 cases)
[PASSED] nnstreamer_decoder_pose (2 passed among 2 cases)
[PASSED] nnstreamer_repo (66 passed among 66 cases)
[PASSED] nnstreamer_decoder_image_labeling (11 passed among 11 cases)
[PASSED] transform_dimchg (5 passed among 5 cases)
[PASSED] transform_arithmetic (34 passed among 34 cases)
[PASSED] codegen (12 passed among 12 cases)
[PASSED] nnstreamer_decoder (17 passed among 17 cases)
[PASSED] nnstreamer_filter_pytorch (7 passed among 7 cases)
[PASSED] nnstreamer_filter_tensorflow (4 passed among 4 cases)
[PASSED] nnstreamer_repo_dynamicity (10 passed among 10 cases)
[PASSED] nnstreamer_merge (89 passed among 89 cases)
[PASSED] nnstreamer_demux (37 passed among 37 cases)
[PASSED] nnstreamer_filter_custom (31 passed among 31 cases)
[PASSED] nnstreamer_filter_reload (2 passed among 2 cases)
[PASSED] nnstreamer_repo_lstm (2 passed among 2 cases)
[PASSED] transform_transpose (7 passed among 7 cases)
[PASSED] transform_typecast (20 passed among 20 cases)
==================================================
[PASSED] All Test Groups (27) Passed!
         TC Passed: 533 / Failed: 0 / Ignored: 0 
```
#### Example 1: pose-estimation-tflite (via videotestsrc): Works
#### Example 2: object-detection-tflite  (via videotestsrc): Fails
```bash
INFO: Initialized TensorFlow Lite runtime.
Setting pipeline to PAUSED ...
Pipeline is PREROLLING ...
Caught SIGSEGV
Caught SIGSEGV
./gst-launch-object-detection-tflite.sh: line 9: 20326 Segmentation fault      gst-launch-1.0 videotestsrc name=cam_src ! videoconvert ! videoscale ! video/x-raw,width=640,height=480,format=RGB,framerate=30/1 ! tee name=t t. ! queue leaky=2 max-size-buffers=2 ! videoscale ! video/x-raw,width=300,height=300,format=RGB ! tensor_converter ! tensor_transform mode=arithmetic option=typecast:float32,add:-127.5,div:127.5 ! tensor_filter framework=tensorflow-lite model=tflite_model/ssd_mobilenet_v2_coco.tflite ! tensor_decoder mode=bounding_boxes option1=tflite-ssd option2=tflite_model/coco_labels_list.txt option3=tflite_model/box_priors.txt option4=640:480 option5=300:300 ! compositor name=mix sink_0::zorder=2 sink_1::zorder=1 ! videoconvert ! ximagesink t. ! queue leaky=2 max-size-buffers=10 ! mix.
```
#### Example 3: speech-command-tflite  (via audiotestsrc): Works

## After applying this PR,
#### Unit Test Cases
```bash
./tests $ ssat
...omission...
==================================================
[PASSED] nnstreamer_converter (28 passed among 28 cases)
[PASSED] nnstreamer_filter_caffe2 (2 passed among 2 cases)
[PASSED] nnstreamer_mux (84 passed among 84 cases)
[PASSED] nnstreamer_filter_tensorflow_lite (23 passed among 23 cases)
[PASSED] transform_stand (3 passed among 3 cases)
[PASSED] nnstreamer_decoder_image_segment (4 passed among 4 cases)
[PASSED] nnstreamer_split (21 passed among 21 cases)
[PASSED] nnstreamer_decoder_boundingbox (6 passed among 6 cases)
[PASSED] nnstreamer_filter_python (6 passed among 6 cases)
[PASSED] nnstreamer_repo_rnn (2 passed among 2 cases)
[PASSED] nnstreamer_decoder_pose (2 passed among 2 cases)
[PASSED] nnstreamer_repo (66 passed among 66 cases)
[PASSED] nnstreamer_decoder_image_labeling (11 passed among 11 cases)
[PASSED] transform_dimchg (5 passed among 5 cases)
[PASSED] transform_arithmetic (34 passed among 34 cases)
[PASSED] codegen (12 passed among 12 cases)
[PASSED] nnstreamer_decoder (17 passed among 17 cases)
[PASSED] nnstreamer_filter_pytorch (7 passed among 7 cases)
[PASSED] nnstreamer_filter_tensorflow (4 passed among 4 cases)
[PASSED] nnstreamer_repo_dynamicity (10 passed among 10 cases)
[PASSED] nnstreamer_merge (89 passed among 89 cases)
[PASSED] nnstreamer_demux (37 passed among 37 cases)
[PASSED] nnstreamer_filter_custom (31 passed among 31 cases)
[PASSED] nnstreamer_filter_reload (2 passed among 2 cases)
[PASSED] nnstreamer_repo_lstm (2 passed among 2 cases)
[PASSED] transform_transpose (7 passed among 7 cases)
[PASSED] transform_typecast (20 passed among 20 cases)
==================================================
[PASSED] All Test Groups (27) Passed!
         TC Passed: 535 / Failed: 0 / Ignored: 0 
```
#### Example 1: pose-estimation-tflite (via videotestsrc): Works
#### Example 2: object-detection-tflite  (via videotestsrc): Works but ximagesink freezes
#### Example 3: speech-command-tflite  (via audiotestsrc): Works

**Comment by anyj0527** [comment]
@myungjoo, @wooksong Thanks for reporting.

I am trying to reduce the test model binary (related to #2169) , but too simple model does not make the regression. I think the run-time memory environment is related to the issue.

**Comment by anyj0527** [comment]
PR updated: Modify the testcase number correctly.

Regard #2169: After some experiments, I found that small model whose file size is less than 3.X MB does not show this issue. There might be a efficient way to check this issue without a big model, but I can't devise it now..

**Comment by myungjoo** [comment]
@helloahn's concern is resolved. Merging.

**Pull request merged**